### PR TITLE
Update GeofenceEditMap.js

### PR DESCRIPTION
**Pull request by Aliber009**
Adding a full draw control to mapbox-gl-draw plugin , with all the features , Point , polygon, and linestring handling all events (read comments above)

### CODE REVIEW

File: modern/src/map/GeofenceEditMap.js [hunk 1]
```diff
@@ -1,24 +1,559 @@
-import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css'
-import MapboxDraw from '@mapbox/mapbox-gl-draw';
-import { useEffect } from 'react';
-
+import * as React from 'react';
+import { MapContext } from '@urbica/react-map-gl';
+import MapboxDraw from 'C:/Users/ALTERNATOR/Desktop/Rapport-PFE/PFE/Traccar/nextTrac/modern/node_modules/@mapbox/mapbox-gl-draw';
```
Comment (tananaev) [inline modern/src/map/GeofenceEditMap.js:3]: This needs to be fixed.

File: modern/src/map/GeofenceEditMap.js [hunk 2]
```diff
@@ -1,24 +1,559 @@
-import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css'
-import MapboxDraw from '@mapbox/mapbox-gl-draw';
-import { useEffect } from 'react';
-
+import * as React from 'react';
```
Comment (tananaev) [inline modern/src/map/GeofenceEditMap.js:1]: Please don't use star imports.

File: modern/src/map/GeofenceEditMap.js [hunk 3]
```diff
@@ -1,24 +1,559 @@
...
+import { MapContext } from '@urbica/react-map-gl';
+import MapboxDraw from 'C:/Users/ALTERNATOR/Desktop/Rapport-PFE/PFE/Traccar/nextTrac/modern/node_modules/@mapbox/mapbox-gl-draw';
+import theme from '@mapbox/mapbox-gl-draw/src/lib/theme';
 import { map } from './Map';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css';
+import {useState} from 'react';
```
Comment (tananaev) [inline modern/src/map/GeofenceEditMap.js:8]:
```suggestion
import { useState } from 'react';
```

**Review by tananaev — COMMENTED** [review]
This pull request doesn't seem to be in a ready state. Formatting is broken, "fixme" comments all over the place. I also see a lot of unused parameters. Once it's ready, please let me know.

File: modern/src/map/GeofenceEditMap.js (hunk 3 above)
Comment (Aliber009) [reply modern/src/map/GeofenceEditMap.js:8]: it s working i just left those comments there, i will fomat it now

**Comment by tananaev** [comment]
Polylines are already supported on master.

**Pull request closed without merging**